In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
df_info = pd.read_csv("../data/processed/info.csv")
df_hospitales = pd.read_csv("../data/processed/hospitales.csv")
df_grp = pd.read_csv("../data/processed/grd.csv")


In [3]:
df_hospitales.head()

Nombre Servicio de Salud  \
0  Metropolitano Norte                           ...   
1  Metropolitano Occidente                       ...   
2  Metropolitano Occidente                       ...   
3  Metropolitano Occidente                       ...   
4  Metropolitano Occidente                       ...   

                         Nombre Establecimiento  \
0                Complejo Hospitalario San José   
1        Hospital Adalberto Steeger (Talagante)   
2               Hospital Dr. Félix Bulnes Cerda   
3                     Hospital San Juan de Dios   
4  Instituto Traumatológico Dr. Teodoro Gebauer   

   Área Cuidados Intensivos Adultos  Área Cuidados Intermedios Adulto  \
0                                11                                33   
1                                 6                                11   
2                                 3                                 6   
3                                29                                47   
4                                12                                 6   

   Total Camas Críticas  Máxima cantidad de camas a aumetar  \
0                    44                                  10   
1                    17                                   5   
2                     9                                   5   
3                    76                                  30   
4                    18                                   5   

   Costo de instalar una cama [MM$]  \
0                              1407   
1                              1263   
2                              1310   
3                              1066   
4                              1433   

   Costo de operación anual de una cama [MM$]  Costo diario GRD1 [MM$]  \
0                                        47.2                    0.593   
1                                        62.1                    0.493   
2                                        65.9                    0.580   
3                                        39.2                    0.482   
4                                        55.5                    0.407   

   Costo diario GRD2 [MM$]  Costo diario GRD3 [MM$]  Costo diario GRD4 [MM$]  \
0                    0.509                    0.367                    0.492   
1                    0.357                    0.429                    0.382   
2                    0.325                    0.375                    0.409   
3                    0.554                    0.564                    0.331   
4                    0.449                    0.468                    0.433   

   Costo diario GRD5 [MM$]  Costo diario GRD6 [MM$]  Costo diario GRD7 [MM$]  \
0                    0.524                    0.405                    0.571   
1                    0.469                    0.386                    0.543   
2                    0.561                    0.362                    0.328   
3                    0.438                    0.509                    0.544   
4                    0.573                    0.392                    0.409   

   Costo diario GRD8 [MM$]  Costo diario GRD9 [MM$]  Costo diario GRD10 [MM$]  
0                    0.474                    0.498                     0.305  
1                    0.362                    0.564                     0.333  
2                    0.314                    0.536                     0.436  
3                    0.410                    0.441                     0.359  
4                    0.391                    0.418                     0.334

In [4]:
df_hospitales.columns = df_hospitales.columns.str.strip()

In [5]:
p3 = df_hospitales[["Nombre Establecimiento","Área Cuidados Intensivos Adultos", "Área Cuidados Intermedios Adulto"]]
p3

Nombre Establecimiento  \
0                      Complejo Hospitalario San José   
1              Hospital Adalberto Steeger (Talagante)   
2                     Hospital Dr. Félix Bulnes Cerda   
3                           Hospital San Juan de Dios   
4        Instituto Traumatológico Dr. Teodoro Gebauer   
5                 Hospital Clínico San Borja-Arriarán   
6   Hospital de Urgencia Asistencia Pública Dr. Al...   
7                                  Hospital el Carmen   
8                               Hospital Del Salvador   
9                          Hospital Dr. Luis Tisné B.   
10       Instituto de Neurocirugía Dr. Alfonso Asenjo   
11  Instituto Nacional de Enfermedades Respiratori...   
12                       Hospital Barros Luco Trudeau   
13  Hospital de Enfermedades Infecciosas Dr. Lucio...   
14              Hospital Dr. Exequiel González Cortés   
15                                   Hospital El Pino   
16          Hospital Parroquial de San Bernardo         
17         Complejo Hospitalario Dr. Sótero del Río     
18                     Hospital Clínico de la Florida   
19              Hospital Metropolitano ( ex Militar )   
20         Hospital Padre Alberto Hurtado (San Ramón)   

    Área Cuidados Intensivos Adultos  Área Cuidados Intermedios Adulto  
0                                 11                                33  
1                                  6                                11  
2                                  3                                 6  
3                                 29                                47  
4                                 12                                 6  
5                                 17                                26  
6                                 24                                20  
7                                  6                                12  
8                                 22                                24  
9                                  8                                16  
10                                12                                11  
11                                15                                29  
12                                31                                43  
13                                11                                 9  
14                                23                                 8  
15                                12                                20  
16                                 6                                10  
17                                20                                35  
18                                 9                                15  
19                                12                                20  
20                                 6                                10

In [15]:
p3.loc[:, "intensive>intermediate"] = np.where(
    p3["Área Cuidados Intensivos Adultos"] > p3["Área Cuidados Intermedios Adulto"],
    1,
    0
)


In [16]:
p3.head(7)

Nombre Establecimiento  \
0                     Complejo Hospitalario San José   
1             Hospital Adalberto Steeger (Talagante)   
2                    Hospital Dr. Félix Bulnes Cerda   
3                          Hospital San Juan de Dios   
4       Instituto Traumatológico Dr. Teodoro Gebauer   
5                Hospital Clínico San Borja-Arriarán   
6  Hospital de Urgencia Asistencia Pública Dr. Al...   

   Área Cuidados Intensivos Adultos  Área Cuidados Intermedios Adulto  \
0                                11                                33   
1                                 6                                11   
2                                 3                                 6   
3                                29                                47   
4                                12                                 6   
5                                17                                26   
6                                24                                20   

   intensive>intermediate  
0                       0  
1                       0  
2                       0  
3                       0  
4                       1  
5                       0  
6                       1

In [17]:
p3.loc[p3["intensive>intermediate"] == 1]

Nombre Establecimiento  \
4        Instituto Traumatológico Dr. Teodoro Gebauer   
6   Hospital de Urgencia Asistencia Pública Dr. Al...   
10       Instituto de Neurocirugía Dr. Alfonso Asenjo   
13  Hospital de Enfermedades Infecciosas Dr. Lucio...   
14              Hospital Dr. Exequiel González Cortés   

    Área Cuidados Intensivos Adultos  Área Cuidados Intermedios Adulto  \
4                                 12                                 6   
6                                 24                                20   
10                                12                                11   
13                                11                                 9   
14                                23                                 8   

    intensive>intermediate  
4                        1  
6                        1  
10                       1  
13                       1  
14                       1

# Evaluacion 4

1. ¿Cuál de las siguientes alternativas corresponde a la estimación de la cantidad de pacientes con GRD9 que llegarán el día 425 al Hospital Salvador utilizando un método de media móvil sobre los últimos 5 días (días 420 a 424) redondeado al entero superior?  

In [ ]:
df_info.head()

Unnamed: 0  Número de llegada  \
0           0                  1   
1           1                  2   
2           2                  3   
3           3                  4   
4           4                  5   

                          Hospital al que llega    GRD  Día de llegada  \
0  Instituto Traumatológico Dr. Teodoro Gebauer   GRD3               1   
1           Hospital Clínico San Borja-Arriarán   GRD7               1   
2                    Hospital Dr. Luis Tisné B.   GRD7               1   
3         Hospital Dr. Exequiel González Cortés  GRD10               1   
4                  Hospital Barros Luco Trudeau   GRD6               1   

   Día de salida  Estadia  
0              4        3  
1              5        4  
2              5        4  
3             11       10  
4              6        5

In [23]:
df_info.loc[
    (df_info["Día de llegada"] >= 420) &
    (df_info["GRD"] >= "GRD9") &
    (df_info["Hospital al que llega"] == "Hospital Del Salvador")
]

Unnamed: 0  Número de llegada  Hospital al que llega   GRD  \
64793       64793              64794  Hospital Del Salvador  GRD9   
64859       64859              64860  Hospital Del Salvador  GRD9   
64861       64861              64862  Hospital Del Salvador  GRD9   
65049       65049              65050  Hospital Del Salvador  GRD9   
65066       65066              65067  Hospital Del Salvador  GRD9   
65160       65160              65161  Hospital Del Salvador  GRD9   
65179       65179              65180  Hospital Del Salvador  GRD9   
65182       65182              65183  Hospital Del Salvador  GRD9   
65289       65289              65290  Hospital Del Salvador  GRD9   

       Día de llegada  Día de salida  Estadia  
64793             420            425        5  
64859             420            425        5  
64861             420            425        5  
65049             421            426        5  
65066             421            426        5  
65160             422            427        5  
65179             422            427        5  
65182             422            427        5  
65289             422            427        5

In [32]:
df_info.loc[
    (df_info["Día de llegada"] >= 420) &
    (df_info["GRD"] >= "GRD9") &
    (df_info["Hospital al que llega"] == "Hospital Del Salvador")
][["Día de llegada", "Número de llegada"]].groupby("Día de llegada").count()

Número de llegada
Día de llegada                   
420                             3
421                             2
422                             4

In [33]:
df_info.loc[
    (df_info["Día de llegada"] >= 420) &
    (df_info["GRD"] >= "GRD9") &
    (df_info["Hospital al que llega"] == "Hospital Del Salvador")
][["Día de llegada", "Número de llegada"]].groupby("Día de llegada").count().mean()

Número de llegada    3.0
dtype: float64

2. ¿Cuál de las siguientes alternativas corresponde a la estimación de la cantidad de pacientes que llegarán el día 425 al Hospital Salvador utilizando un método de media móvil sobre los últimos 5 días (días 420 a 424) redondeado al entero superior?  

In [39]:
df_info.loc[
    (df_info["Día de llegada"] >= 420) &
    (df_info["Hospital al que llega"] == "Hospital Del Salvador")
][["Día de llegada", "Número de llegada"]].groupby("Día de llegada").count().mean().round()

Número de llegada    12.0
dtype: float64

3. ¿Cuál de las siguientes alternativas corresponde a la estimación de la cantidad de pacientes que llegarán el día 425 a la red utilizando un método de media móvil sobre los últimos 5 días (días 420 a 424) redondeado al entero superior?   

In [38]:
df_info.loc[
    (df_info["Día de llegada"] >= 420)
][["Día de llegada", "Número de llegada"]].groupby("Día de llegada").count().mean().round()

Número de llegada    149.0
dtype: float64

4. ¿Cuál de las siguientes alternativas corresponde a la estimación de la cantidad de pacientes con GRD9 que llegarán el día 425 al Hospital Salvador utilizando una regresión lineal simple sobre los últimos 5 días (días 420 a 424) redondeado al entero superior?  

In [41]:
data = df_info.loc[
    (df_info["Día de llegada"] >= 420) &
    (df_info["GRD"] >= "GRD9") &
    (df_info["Hospital al que llega"] == "Hospital Del Salvador")
][["Día de llegada", "Número de llegada"]].groupby("Día de llegada").count()
data

Número de llegada
Día de llegada                   
420                             3
421                             2
422                             4

In [43]:
data_reset = data.reset_index()
data_reset

Día de llegada  Número de llegada
0             420                  3
1             421                  2
2             422                  4

In [44]:
X = data_reset[["Día de llegada"]]
y = data_reset["Número de llegada"] 

In [48]:
from sklearn.linear_model import LinearRegression

In [49]:
model = LinearRegression()
model.fit(X, y)

LinearRegression()

In [54]:
prediccion = model.predict([[425]])

/Users/jasonsepulveda/Library/CloudStorage/OneDrive-UniversidadCatólicadeChile/master_data_science/2025/01_python_ciencia_datos/dev/.venv/lib/python3.12/site-packages/sklearn/utils/validation.py:2739: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(


In [55]:
prediccion

array([5.])

5. ¿Cuál de las siguientes alternativas corresponde a la estimación de la cantidad de pacientes que llegarán el día 425 al Hospital Salvador utilizando una regresión lineal simple sobre los últimos 5 días (días 420 a 424) redondeado al entero superior? 

In [56]:
data = df_info.loc[
    (df_info["Día de llegada"] >= 420) &
    (df_info["Hospital al que llega"] == "Hospital Del Salvador")
][["Día de llegada", "Número de llegada"]].groupby("Día de llegada").count()
data

Número de llegada
Día de llegada                   
420                            14
421                            12
422                            18
423                             9
424                             6

In [57]:
data_reset = data.reset_index()
data_reset

Día de llegada  Número de llegada
0             420                 14
1             421                 12
2             422                 18
3             423                  9
4             424                  6

In [58]:
X = data_reset[["Día de llegada"]]
y = data_reset["Número de llegada"] 

In [61]:
model = LinearRegression()
model.fit(X, y)
prediccion = model.predict([[425]]).round()

/Users/jasonsepulveda/Library/CloudStorage/OneDrive-UniversidadCatólicadeChile/master_data_science/2025/01_python_ciencia_datos/dev/.venv/lib/python3.12/site-packages/sklearn/utils/validation.py:2739: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(


In [62]:
prediccion

array([6.])

6. ¿Cuál de las siguientes alternativas corresponde a la estimación de la cantidad de pacientes que llegarán a la red el día 425 utilizando una regresión lineal simple sobre los últimos 5 días (días 420 a 424) redondeado al entero superior?  

In [65]:
data = df_info.loc[
    (df_info["Día de llegada"] >= 420)
][["Día de llegada", "Número de llegada"]].groupby("Día de llegada").count()
data

Número de llegada
Día de llegada                   
420                           183
421                           159
422                           160
423                           159
424                            82

In [66]:
data_reset = data.reset_index()
data_reset

Día de llegada  Número de llegada
0             420                183
1             421                159
2             422                160
3             423                159
4             424                 82

In [67]:
X = data_reset[["Día de llegada"]]
y = data_reset["Número de llegada"] 

model = LinearRegression()
model.fit(X, y)
prediccion = model.predict([[425]]).round()
prediccion

/Users/jasonsepulveda/Library/CloudStorage/OneDrive-UniversidadCatólicadeChile/master_data_science/2025/01_python_ciencia_datos/dev/.venv/lib/python3.12/site-packages/sklearn/utils/validation.py:2739: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(


array([88.])

7. ¿Cuál de las siguientes alternativas corresponde al valor del coeficiente R2 (R cuadrado) de una estimación sobre la cantidad de pacientes que llegarán a la red el día 425 utilizando una regresión lineal simple sobre los últimos 5 días (días 420 a 424)?  

In [68]:
r2 = model.score(X, y)
r2

0.6840340642392606

8. Usted realizó una estimación sobre el total de pacientes que llegarán a la red los días 420 a 424, siendo esta predicción: 156, 123, 178, 200 y 155, para los días 420 a 424 respectivamente. ¿Cuál de las siguientes alternativas corresponde al error total de esta estimación?  

In [71]:
predictions = np.array([156, 123, 178, 200, 155])
actuals = np.array([183, 159, 160, 159, 82])
errors = actuals - predictions
errors.sum()

np.int64(-69)

9. Los valores del intercepto y la pendiente de un modelo de regresión lineal para predecir la cantidad de pacientes que llegarán un día cualquiera al Hospital San José son 60 y 10 respectivamente. La variable independiente de esta regresión corresponde a las llegadas del hace 7 días. 

¿Cuál de las siguientes alternativas corresponde a la estimación de esta regresión para los días 422, 423 y 424?

In [80]:
data = df_info.loc[
    (df_info["Día de llegada"] >= 414) &
    (df_info["Hospital al que llega"] == "Complejo Hospitalario San José")
][["Día de llegada", "Número de llegada"]].groupby("Día de llegada").count()
data

Número de llegada
Día de llegada                   
414                            13
415                            11
416                             7
417                            11
418                            12
419                             7
420                            12
421                            14
422                            15
423                            10
424                             7

In [81]:
# Model -> y = wx + b
b = 60
w = 10
x_422 = 11
x_423 = 7
x_424 = 11

y_422 = w * x_422 + b
y_423 = w * x_423 + b
y_424 = w * x_424 + b

print(y_422, y_423, y_424)

170 130 170
